barman/retention_policies.py

```diff
@@ -126,6 +126,40 @@ def to_json(self):
         """
         return "%s %s %s" % (self.mode, self.value, self.unit)
 
+    def propagate_annotation_to_children(self, backup_info, report, annotation):
+        # If backup is postgres with incremental backup children
+        # create tree o backups and make them all OBSOLETE.
+        msg = "No children to propagate annotation..."
+        backup_id_tree = [
+            backup.backup_id for backup in list(backup_info.walk_backups_tree())
+        ][:-1]
+        for _id in backup_id_tree:
+            report[_id] = annotation
+        len_format = 2 if len(backup_id_tree) <= 7 else 3
+        msg = "Propagating %s annotation of backup %s to n=%s children %s." % (
+            annotation,
+            backup_info.backup_id,
+            str(len(backup_id_tree)),
+            ",".join(backup_id_tree[:len_format])
+            + "..."
+            + ",".join(backup_id_tree[-len_format:]),
+        )
+        _logger.debug(msg)
+
+    def has_children(self, backup_info):
+        return (
+            isinstance(backup_info, BackupInfo)
```

Comment:
I used all this checks because there were a lot of tests braking in the TestCloudBackupDelete class if they are not used. Took a lot of time to find that this was the problem braking the tests.